File: src/test/java/voidchess/figures/CastlingFigureTest.java
package voidchess.figures;

import org.testng.annotations.Test;
import voidchess.board.move.Move;
import voidchess.board.move.Position;

import static org.testng.Assert.assertEquals;

public class CastlingFigureTest {
    @Test
    public void testGetInitialPosition() {
        Position initial = Position.byCode("e1");
        Position second = Position.byCode("d1");

        King king = new King(true, initial);

        assertEquals(king.getInitialPosition(), initial);
        king.figureMoved(Move.get(initial, second));
        assertEquals(king.getInitialPosition(), initial);
    }
}

File: src/test/kotlin/voidchess/figures/CastlingFigureTest.kt
package voidchess.figures

import org.testng.annotations.Test
import voidchess.board.move.Move
import voidchess.board.move.Position
import kotlin.test.assertEquals

class CastlingFigureTest {
    @Test
    fun testGetInitialPosition() {
        val initial = Position.byCode("e1")
        val second = Position.byCode("d1")

        val king = King(true, initial)

        assertEquals(initial, king.initialPosition)
        king.figureMoved(Move[initial, second])
        assertEquals(initial, king.initialPosition)
    }
}
